src/ipc_listener.cpp

```diff
@@ -208,9 +260,13 @@ int zmq::ipc_listener_t::set_address (const char *addr_)
 
 int zmq::ipc_listener_t::close ()
 {
-    zmq_assert (s != retired_fd);
-    int rc = ::close (s);
-    errno_assert (rc == 0);
+    int rc;
+
+    if ( s != -1 ) {
+      zmq_assert (s != retired_fd);
```

Comment:
On Mon, Mar 14, 2016 at 9:08 AM, Luca Boccassi <notifications@github.com> wrote:
>
> In src/ipc_listener.cpp:
>
> > @@ -208,9 +260,13 @@ int zmq::ipc_listener_t::set_address (const char *addr_)
> >
> >  int zmq::ipc_listener_t::close ()
> >  {
> > -    zmq_assert (s != retired_fd);
> > -    int rc = ::close (s);
> > -    errno_assert (rc == 0);
> > +    int rc;
> > +
> > +    if ( s != -1 ) {
> > +      zmq_assert (s != retired_fd);
>
> retired_fd is -1, so this check doesn't make much sense. Why the added "if {}" block?

I'm trying to handle the case where the directory for the socket
failed to be created (i.e. mkdtemp() failed), then s would remain -1.
This would cause the assertion error "s != retired_fd" and casue
::close to return an error with errno set to EBADF.  Though I suppose
this is also the case if open_socket() failed as well.  So it seems
that in the case of either mkdtemp() or open_socket() failing, we
don't want the assertion or ::close() failure here.

Pete